src/utils/hooks/useGetVideos.js

```diff
@@ -1,75 +1,86 @@
-import { useContext } from 'react';
+import { useContext, useCallback } from 'react';
 import { StoreContext } from '../store/StoreContext';
 
 const useGetVideos = () => {
   const { dispatch } = useContext(StoreContext);
 
-  // const API_KEY = process.env.REACT_APP_YOUTUBE_APIKEY;
-  const API_KEY = 'AIzaSyDRKJLorYEYkVTc9kfMowzFCCSBaXMDAK8';
+  const API_KEY = process.env.REACT_APP_YOUTUBE_APIKEY;
   const API_URL_TEST = 'mockService.json';
   const BASE_API_URL = `https://youtube.googleapis.com/youtube/v3/search?part=snippet&type=video&videoDuration=medium&videoCategoryId=10&maxResults=20&key=${API_KEY}`;
-  const testingAPI = true; //Only for testing purposes set "false" to test with youtube API
+  const testingAPI = false; //Only for testing purposes set "false" to test with youtube API
 
-  const fetchVideos = async (requestParam, type) => {
-    const API_URL = `${BASE_API_URL}${requestParam}`;
-    const URL = !testingAPI ? API_URL : API_URL_TEST;
+  const fetchVideos = useCallback(
```

Comment:
I'm glad you find out how to solve the problem you had, sorry for not being able to help you review this earlier.